core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -2239,14 +2239,34 @@ public void setNodes(final List<? extends Node> n) throws IOException {
      * but we also call this periodically to self-heal any data out-of-sync issue.
      */
     /*package*/ void trimLabels() {
+        trimLabels((Set) null);
```

Comment:
would be cleaner as `trimLabels(eveyLabelInJenkinsAgentOrCloud)` but I understand the performance shortcut is desired here.